source/common/http/conn_manager_impl.cc

```diff
@@ -564,28 +564,35 @@ void ConnectionManagerImpl::ActiveStream::traceRequest() {
   if (cached_route_.value() && cached_route_.value()->decorator()) {
     cached_route_.value()->decorator()->apply(*active_span_);
 
+    // Cache decorated operation.
+    if (!cached_route_.value()->decorator()->getOperation().empty()) {
```

Comment:
Technically, the route can change during filter iteration. This problem was partly here already. The `apply()` call on line 565 I think should probably happen at span finish time (I noticed this recently)? In the case of the cached decorator header, I think technically if  `clearRouteCache()` is called, we should refresh this on route reload, which makes me wonder if this code should somehow move into the route cache load code. In any case, I think this is too complicated of an issue to solve in this PR. Can you throw a TODO in here that will remind us that this is probably broken in the case a filter changes the route?

Reply:
Done - only thing to consider if moving where the decorator is applied is setting the x-envoy-decorator-operation in a forwarded request (i.e. for a client span).